packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts

```diff
@@ -0,0 +1,80 @@
+import * as path from 'path';
+import fs from 'fs-extra';
+import {
+  $TSAny, JSONUtilities, pathManager, stateManager,
+} from 'amplify-cli-core';
+
+const {
+  readJson,
+} = JSONUtilities;
+
+/**
+ * Updates Cognito files that are tracked so that the diff is detected for an `amplify push`
+ */
+export const updateCognitoTrackedFiles = async (): Promise<void> => {
+  const currentCloudBackendDir = pathManager.getCurrentCloudBackendDirPath();
+  const localBackendDir = pathManager.getBackendDirPath();
+  const amplifyMeta = stateManager.getMeta();
+  const cognitoResource = stateManager.getResourceFromMeta(amplifyMeta, 'auth', 'Cognito', undefined, false);
+
+  if (!fs.existsSync(currentCloudBackendDir) || !cognitoResource) {
+    return;
+  }
+
+  const { resourceName } = cognitoResource;
+
+  if (await detectCognitoDiff(currentCloudBackendDir, localBackendDir, resourceName)) {
+    await addExtraLineToCliInputsJson(localBackendDir, resourceName);
```

Comment:
if the goal is to trigger change detection on the CLIInputsJSON file, it may help to rename this "triggerChangeDetectionOnCLIInputsJSON"; then leave implementation details out. Implementations change, but intention is more resilient

Reply:
resolving as style suggestion